qiskit_nature/utils/testing.py

```diff
@@ -0,0 +1,51 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Testing utilities."""
+
+from typing import Any
+
+import numpy as np
+
+
+# TODO see if type of seed can be specified instead of using Any
+def parse_random_seed(seed: Any) -> np.random.Generator:
+    """Parse a random number generator seed and return a Generator.
+
+    Args:
+        seed: The pseudorandom number generator or seed. Should be an
+            instance of `np.random.Generator` or else a valid input to
+            `np.random.default_rng`
+
+    Returns:
+        The np.random.Generator instance
+    """
+    if isinstance(seed, np.random.Generator):
+        return seed
+    return np.random.default_rng(seed)
+
+
+def random_antisymmetric_matrix(dim: int, seed: Any = None) -> np.ndarray:
+    """Return a random antisymmetric matrix.
+
+    Args:
+        dim: The width and height of the matrix.
+        seed: The pseudorandom number generator or seed. Should be an
+            instance of `np.random.Generator` or else a valid input to
+            `np.random.default_rng`
+
+    Returns:
+        The sampled antisymmetric matrix.
+    """
+    rng = parse_random_seed(seed)
+    mat = rng.standard_normal((dim, dim)) + 1j * rng.standard_normal((dim, dim))
```

Comment:
Since this is under testing I can't see the rationale for the complexity here - it makes more seeds etc for the user to deal with to get things reproducible. I suggested using the algorithm_globals random generator for consistency with code elsewhere. It would be random by default but can be seeded. Its something users already are accustomed to using across the repos here. Having some exposed public API `parse_random_seed`, whose function is not immediately apparent by name is just adding to users complexity. If you feel that this matrix method truly needs to have an option to be independent of the common generator that we use, I would suggest  rather than pass seed in let the user pass a generator. And by default if None, use algorithm globals. Though honestly my preference would be to keep it simple and not have that option, ie. just use algorithm globals - it can always be added later in a backwards compatible way should it ever be reqd.

Reply:
> If you feel that this matrix method truly needs to have an option to be independent of the common generator that we use

I truly feel that the correct way to implement this function is to allow the seed to be passed. This allows the function behavior to be reproduced independently of some global state. The closest analogues of this function in Qiskit are the functions in [`qiskit.quantum_info.operators.random`](https://github.com/Qiskit/qiskit-terra/blob/89c4f6eb5c56f49ec5149e5be39c77d36b182b4e/qiskit/quantum_info/operators/random.py), and I have made my implementation consistent with that.

> I would suggest rather than pass seed in let the user pass a generator.

 The seed _is_ allowed to be a generator. Again, this is the same behavior as [`qiskit.quantum_info.operators.random`](https://github.com/Qiskit/qiskit-terra/blob/89c4f6eb5c56f49ec5149e5be39c77d36b182b4e/qiskit/quantum_info/operators/random.py)

> Having some exposed public API parse_random_seed, whose function is not immediately apparent by name is just adding to users complexity.

Ok, I'll remove it from the public API.

Reply:
> I truly feel that the correct way to implement this function is to allow the seed to be passed. 

Ok, what you linked defines its own global level default rng in that module and all methods use that by default. Otherwise they use a generator as passed or build a local one using the seed. I can accept an implementation done the same way though I guess I question the notion of calling it a testing module where to me potentially that degree of independence would be less important.

Reply:
Done.